lib/CLI/Parser.hs

```diff
@@ -568,6 +549,41 @@ parseFieldNameWhile whileCond = do
 parseFieldContentsEof :: MParser FieldContents
 parseFieldContentsEof = FieldContents . T.pack <$> P.manyTill P.anySingle P.eof
 
+
+parseSort :: MParser (Sort, Direction)
+parseSort = do
+  sort <- parseSortMeans
+  void $ P.char ':'
+  direction <- parseSortDirection
+  return (sort, direction)
+
+
+parseSortDirection :: MParser Direction
+parseSortDirection =
+  try (P.string "asc" $> Asc)
+  <|> P.string "desc" $> Desc
+
+parseSortMeans :: MParser Sort
+parseSortMeans =
+  try parseSortMeansByNameOrDate
+  <|> parseSortMeansByFieldName
+
+parseSortMeansByFieldName :: MParser Sort
+parseSortMeansByFieldName = do
+  fieldName <- parseFieldNameWhile (/= ':')
+  void $ P.char ':'
+  P.choice (
+    [ (SortByFieldDate fieldName)     <$ (P.string "date")
+    , (SortByFieldContents fieldName) <$ (P.string "contents") ] :: [MParser Sort] )
```

Comment:
You can avoid using parentheses like this
```suggestion
  P.choice @[] @_ @Sort $
    [ (SortByFieldDate fieldName)     <$ (P.string "date")
    , (SortByFieldContents fieldName) <$ (P.string "contents")
    ]
```
Btw, in list declarations we put a trailing `]` (see [code-style](https://github.com/serokell/coffer/blob/main/docs/code-style.md#list-declarations))